azurerm_eventhub/main.tf

```diff
@@ -0,0 +1,63 @@
+provider "azurerm" {
+  version = "~>1.42"
+}
+
+terraform {
+  # The configuration for this backend will be filled in by Terragrunt
+  backend "azurerm" {}
+}
+
+resource "azurerm_eventhub_namespace" "eventhub_ns" {
+  name                     = local.evhns_resource_name
+  location                 = var.region
+  resource_group_name      = var.resource_group_name
+  sku                      = var.sku
+  capacity                 = var.capacity
+  auto_inflate_enabled     = var.auto_inflate_enabled
+  maximum_throughput_units = var.auto_inflate_enabled ? var.maximum_throughput_units : null
+
+  dynamic "network_rulesets" {
+    for_each = var.network_rulesets == null ? [] : [var.network_rulesets]
```

Comment:
The conditional is not necessary because if network_rulesets is null the for_each step is not executed

Reply:
Thanks @rchessa. 
Here I need the conditional to obtain a list given the network_rulesets **object**. Whether the object is null I want an empty list otherwise a list containing the object itself: [var.network_rulesets].
Do you see a better approach?
If I just go with:

```
for_each = var.network_rulesets
content {
......
```
it does not work since for_each needs a list ...

Reply:
You can declare your object as a list of object and assign [] as a default, in this way you don't need the conditional. I've already used in the [azurerm_cosmosdb_account](https://github.com/pagopa/io-infrastructure-modules-new/blob/master/azurerm_cosmosdb_account/main.tf) module.

Reply:
Thank you are right. I did the change you recommended